modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/DynamicOptionsParameterDeclarationEnricher.java

```diff
@@ -0,0 +1,254 @@
+   * that the correspondent {@link OptionsResolver} will require a config.
+   *
+   * @param modelPropertyBuilder     Options Resolver Model Property Builder
+   * @param parameterizableComponent component to introspect
+   */
+  private void injectConfiguration(OptionsResolverFactoryModelPropertyBuilder modelPropertyBuilder,
+                                   ParameterizableTypeWrapper parameterizableComponent) {
+    List<FieldElement> configFields = parameterizableComponent.getAnnotatedFields(Config.class);
+    if (!configFields.isEmpty()) {
+      FieldElement fieldElement = configFields.get(0);
+      modelPropertyBuilder.withConfig(fieldElement.getMetadataType(typeLoader), fieldElement.getField());
+    }
+  }
+
+  /**
+   * Given a list of {@link ParameterDeclaration}, introspect it and finds all the considered parameters with an associated
+   * {@link OptionsResolver}
+   *
+   * @param parameters parameters to introspect
+   * @return a Map containing all the {@link ParameterDeclaration} with their correspondent {@link OptionsResolver}
+   */
+  private Map<ParameterDeclaration, Class<? extends OptionsResolver>> introspectParameters(List<ParameterDeclaration> parameters) {
+
+    Map<ParameterDeclaration, Class<? extends OptionsResolver>> optionResolverEnabledParameters = new HashMap<>();
+
+    parameters.forEach(param -> getAnnotation(param, DynamicOptions.class)
+        .ifPresent(optionAnnotation -> {
+          optionResolverEnabledParameters.put(param, optionAnnotation.value());
+          param.addModelProperty(new OptionPartModelProperty(getCategory(optionAnnotation)));
+        }));
+
+    return optionResolverEnabledParameters;
+  }
+
+  /**
+   * Given a list of {@link ParameterGroupDeclaration}, introspect it and finds all the considered parameters with an associated
+   * {@link OptionsResolver}
+   *
+   * @param parameterGroups parameter groups to introspect
+   * @return a Map containing all the {@link ParameterGroupDeclaration} with their correspondent {@link OptionsResolver}
+   */
+  private Map<ParameterGroupDeclaration, Class<? extends OptionsResolver>> introspectParameterGroups(List<ParameterGroupDeclaration> parameterGroups) {
+    Map<ParameterGroupDeclaration, Class<? extends OptionsResolver>> optionResolverEnabledParameters = new HashMap<>();
+
+    parameterGroups
+        .forEach(paramGroup -> paramGroup.getModelProperty(ParameterGroupModelProperty.class)
+            .ifPresent(modelProperty -> {
+
+              AnnotatedElement container = modelProperty.getDescriptor().getContainer();
+              if (container != null) {
+                DynamicOptions annotation = container.getAnnotation(DynamicOptions.class);
+                if (annotation != null) {
+
+                  Class<? extends OptionsResolver> value = annotation.value();
+
+                  for (ParameterDeclaration parameterDeclaration : paramGroup.getParameters()) {
```

Comment:
these are the parameters that should not be re-processed in `introspectParameters`